### ISSUE TITLE
[no-extraneous-class] add `allowWithDecorator` option

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#812]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-extraneous-class": "error"
  }
}
```

```TS
@NgModule({
    imports: [CommonModule],
    declarations: [CdkDetailRowDirective],
    exports: [CdkDetailRowDirective]
})
export class DirectivesModule {}
```

**Expected Result**

No errors.

**Actual Result**

Error: `Unexpected empty class.`

**Additional Info**

n/a

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |
